components/cdc/src/endpoint.rs

```diff
@@ -653,45 +669,54 @@ impl<T: 'static + RaftStoreRouter<RocksEngine>> Endpoint<T> {
+                let raft_router_clone = raft_router.clone();
+                async move {
+                    let (tx, rx) = futures03::channel::oneshot::channel();
+                    if let Err(e) = raft_router_clone.significant_send(
+                        region_id,
+                        SignificantMsg::LeaderCallback(Callback::Read(Box::new(move |resp| {
+                            let resp = if !resp.response.get_header().has_error() {
+                                None
+                            } else {
+                                Some(region_id)
+                            };
+                            if tx.send(resp).is_err() {
+                                error!("cdc send tso response failed");
                             }
+                        }))),
+                    ) {
+                        warn!("cdc send LeaderCallback failed"; "err" => ?e, "min_ts" => min_ts);
+                        let deregister = Deregister::Region {
+                            observe_id,
+                            region_id,
+                            err: Error::Request(e.into()),
+                        };
+                        if let Err(e) = scheduler_clone.schedule(Task::Deregister(deregister)) {
+                            error!("schedule cdc task failed"; "error" => ?e);
+                            return None;
                         }
-                    }))),
-                ) {
-                    warn!(
-                        "send LeaderCallback for advancing resolved ts failed";
-                        "err" => ?e,
-                        "min_ts" => min_ts,
-                    );
-                    let deregister = Deregister::Region {
-                        observe_id,
-                        region_id,
-                        err: Error::Request(e.into()),
-                    };
-                    if let Err(e) = scheduler.schedule(Task::Deregister(deregister)) {
-                        error!("schedule cdc task failed"; "error" => ?e);
                     }
+                    rx.await.unwrap_or(None)
                 }
-            }
+            }).collect();
             match scheduler.schedule(Task::RegisterMinTsEvent) {
                 Ok(_) | Err(ScheduleError::Stopped(_)) => (),
                 // Must schedule `RegisterMinTsEvent` event otherwise resolved ts can not
                 // advance normally.
-                Err(err) => panic!(
-                    "failed to schedule regiester min ts event, error: {:?}",
-                    err
-                ),
+                Err(err) => panic!("failed to regiester min ts event, error: {:?}", err),
+            }
+            let resps = futures03::future::join_all(regions).await;
+            let regions = resps
+                .into_iter()
+                .filter_map(|resp| resp)
+                .collect();
+            match scheduler.schedule(Task::MinTS { regions, min_ts }) {
```

Comment:
Please schedule `Task::MinTs` before `TaskRegisterMinTsEvent` to avoid unnecessary concurrent resolved ts events.